src/moonlink_service/src/rest_api.rs

```diff
@@ -161,7 +162,21 @@ pub struct HealthResponse {
     pub timestamp: u64,
 }
 
-/// Create the router with all API endpoints
+/// Map backend error to appropriate HTTP status code based on error type
```

Comment:
It's fine for now to translate via backend error, but I'm not 100% sure that's the case in the future. Subject to change in the future.